### PR TITLE
Buyer should be able to manage a wishlist #58

### LINKED ISSUE #58: Feat: Buyer should be able to manage a wishlist
# User Story: Buyer should be able to manage a wishlist

## Description

As a buyer, i want to view The wishlist management feature enables buyers to create, organize, and maintain a list of desired items they wish to purchase in the future. This provides users with a convenient way to keep track of products they are interested in.

### Acceptance Criteria

- [ ] Buyers can  add products to their wishlist from product pages or search results. 
- [ ] Buyers can view their wishlist at any time to see all the items they have saved. 
- [ ] Buyer can  remove  item from wish list 
- [ ] When buyers are ready to make a purchase, they can easily move items from their wishlist to the shopping cart.

## Developer Notes

- Design and implement the data model for the wishList
- Define the API endpoints and request/response payloads for cart operations
- Integrate with the product catalog service to fetch product details and validate item availability
- Handle edge cases such as out-of-stock items, removed products, or pricing changes
- Implement Validations of product wish List 

## Related Issues

- #37 (Product  CRUD)
- #56 (Product search)

### PR DESCRIPTION
**Pull request by 13XAVI**
# What does this PR do?
This PR is for  user to  view and manage my wishlist so that he/she can keep track of products he/she is interested in purchasing in the future
### Description of Task to be completed?
- [ ] Buyers can  add products to their wishlist from product pages or search results. 
- [ ] Buyers can view their wishlist at any time to see all the items they have saved. 
- [ ] Buyer can  remove  item from wish list.
- [ ]  Implement appropriate error handling and responses  ,unit tests and documentation

### CODE REVIEW

**Review by ambroisegithub — DISMISSED** [review]
It Looks good for merge!!

**Comment by ambroisegithub** [comment]
And For sure you have to  squash your commit(git rebase -i HEAD~N) for keeping have only one commit and then why you skip ci/cd ... otherwise you make great job!!!

File: src/routes/buyer.ts
Comment (jkarenzi) [inline src/routes/buyer.ts:1]: You seem to have 2 route files that handle wishlists.

File: src/routes/buyerRoutes.ts
Comment (jkarenzi) [inline src/routes/buyerRoutes.ts:1]: You seem to have 2 route files that handle wishlists.

File: package.json
Comment (jkarenzi) [inline package.json:1]: why are you modifying these scripts?

**Review by jkarenzi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**